lib/jsonwp-proxy/proxy.js

```diff
@@ -155,19 +158,97 @@ class JWProxy {
     return [res, resBody];
   }
 
+  /**
+   * W3C /timeouts can take as many as 3 timeout types at once, MJSONWP /timeouts only takes one
+   * at a time. So if we're using W3C and proxying to MJSONWP and there's more than one timeout type
+   * provided in the request, we need to do 3 proxies and combine the result
+   *
+   * @param {Object} body Request body
+   * @return {Array} Array of W3C + MJSONWP compatible timeout objects
+   */
+  getTimeoutRequestObjects (body) {
+    const {protocol} = body;
+    const timeoutObjects = [];
+    const baseObj = {
+      script: body.script,
+      pageLoad: body.pageLoad,
+      implicit: body.implicit,
+    };
+    if (protocol === W3C) {
+      for (let [type, ms] of _.toPairs(baseObj)) {
+        if (_.isNumber(ms)) {
+          timeoutObjects.push({
+            ...baseObj,
+            type,
+            ms,
+          });
+        }
+      }
+      return timeoutObjects;
+    }
+
+    return [{
+      ...baseObj,
+      [body.type]: body.ms,
+      type: body.type,
+      ms: body.ms,
+    }];
+  }
+
+  /**
+   * Proxy an array of timeout objects and merge the result
+   * @param {String} url Endpoint url
+   * @param {String} method Endpoint method
+   * @param {Object} body Request body
+   */
+  async proxySetTimeouts (url, method, body) {
+    let response, resBody;
+    for (let timeoutObj of this.getTimeoutRequestObjects(body)) {
+      [response, resBody] = await this.proxy(url, method, timeoutObj);
+
+      let protocol = this.getProtocolFromResBody(resBody);
+
+      // If we got a non-MJSONWP response, return the result, nothing left to do
+      if (protocol !== MJSONWP) {
+        return [response, resBody];
+      }
+
+      // If we got an error, return the error right away
+      if (response.statusCode >= 400) {
+        return [response, resBody];
+      }
+
+      // ...Otherwise, continue to the next timeouts call
+    }
+    return [response, resBody];
+  }
+
+  getProtocolFromResBody (resBody) {
+    if (_.isPlainObject(resBody) && util.hasValue(resBody.status)) {
+      return MJSONWP;
+    } else if (util.hasValue(resBody.value)) {
+      return W3C;
+    }
+  }
+
   async command (url, method, body = null) {
     let response;
     let resBody;
     try {
-      [response, resBody] = await this.proxy(url, method, body);
+      if (url.endsWith('timeouts') && method.toLowerCase() === 'post') {
```

Comment:
maybe `endsWith('/timeouts')`? what if a user is trying to get the 'timeouts' attribute of an element? (i know that's a 'get', just saying...)